v2/kafka-to-gcs/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToGcsFlex.java

```diff
@@ -251,23 +160,20 @@ public static PipelineResult run(KafkaToGcsOptions options) throws UnsupportedOp
   public static void validateAuthOptions(KafkaToGcsOptions options) {
     // Authenticate to Kafka brokers without any auth config. This can be the case when
     // the dataflow pipeline and Kafka broker is on the same network.
-    if (options.getUserNameSecretID().isBlank() && options.getPasswordSecretID().isBlank()) {
-      useKafkaAuth = false;
-      return;
-    }
-
-    if ((options.getUserNameSecretID().isBlank() && !options.getPasswordSecretID().isBlank())
-        || (options.getPasswordSecretID().isBlank() && !options.getUserNameSecretID().isBlank())) {
+    if ((options.getKafkaReadUsernameSecretId().isBlank()
+            && !options.getKafkaReadPasswordSecretId().isBlank())
+        || (options.getKafkaReadPasswordSecretId().isBlank()
+            && !options.getKafkaReadUsernameSecretId().isBlank())) {
       throw new IllegalArgumentException(
           "Both username secret ID and password secret ID should be provided together or left null.");
     }
 
-    if (!SecretVersionName.isParsableFrom(options.getUserNameSecretID())) {
+    if (!SecretVersionName.isParsableFrom(options.getKafkaReadUsernameSecretId())) {
```

Comment:
I think this check will throw the "Provided Secret Username ID must be in the form ..." exception even if the authentication mode doesn't require a username (NONE or SSL).

Reply:
This method was not used. Removed this method.